### ISSUE TITLE
Convert from existing output

### ISSUE THREAD

**Issue by JJRcop** [Tyrrrz/DiscordChatExporter#281]
I'd like to be able to export to one output format and then generate results in the other formats using the exported results as an input, particularly converting to the rich HTML generation from a smaller and easily parseable text format.

It doesn't matter if converting from any format <-> any format is possible, a single supported input format -> any format is acceptable.

This is absolutely a wishlist level feature request, and is not necessary to ever be implemented, just would save a bit from having to hit discord again just to get another format.

**Comment by Jefferderp** [comment]
Currently, the app performs one request per each export file type. Saving as HTML, and then as CSV will force a redownload of the requested channel.

Are you proposing that the app caches each retrieved export, to avoid redownloading? Or that it use an intermediary format like JSON which can be parsed into your desired export type afterwards?

Either way, this would definitely make the process more efficient, and I'd be in favor as well.

**Comment by JJRcop** [comment]
My proposal is a new command or feature that takes an existing exported discord guild or channel(s), meaning you have run the program already and have exported to JSON, Plaintext, or CSV, and convert it to another format you specify.

It's not about caching, just providing the result of a previous export as input.

As a mock example of what it could look like:
```sh
DiscordChatExporter.Cli exportguild -t dEaDBeEf -f json -g 81384788765712384 -o guild.json
DiscordChatExporter.Cli convert -i guild.json -f darkhtml -o guild-pretty.html
```

I don't think it's important that any format be accepted as input, it's fine to choose JSON for example as the format to export to if you want to convert to another format.

**Comment by makeworld-the-better-one** [comment]
> Are you proposing that the app caches each retrieved export, to avoid redownloading?

This would also be cool though, and less suspicious to Discord.

**Comment by Tyrrrz** [comment]
If this will get implemented, it will probably be as part of #265. That way you will be able to export to multiple formats simultaneously without making roundtrips, as I understand that's the main use case here.

Closing in favor of that issue.

**Comment by JJRcop** [comment]
If possible I would still suggest considering this so as to archive a small format (like .json) only and if needed generate the others from it at a later time, instead of archiving the bulkier HTML file too.
